test/integration/targets/ansible-galaxy-collection-scm/tasks/test_invalid_version.yml

```diff
@@ -0,0 +1,18 @@
+- block:
+  - name: test installing a collection with invalid version
+    command: 'ansible-galaxy collection install git+file://{{ test_error_repo_path }}/.git#bad_version_collection -vvvvvv'
+    ignore_errors: yes
+    register: invalid_version_result
+
+  - assert:
+      that:
+        - invalid_version_result is failed
+        - msg in invalid_version_result.stderr
+    vars:
+      req: error_test.bad_version_collection:1.0
+      ver: "1.0 (<class 'float'>)"
```

Comment:
Is it me or does this test skip the code branch with an invalid SemVer value? Are there any pre-existing tests covering that?

Reply:
It's a float, so the `isinstance(float, string_types)` is false. I can add a test with a quoted float to exercise the second fix. 
There are two issues here with unhandled tracebacks.
1) unhandled AttributeError traceback (that's what this tests currently)
2) unhandled ValueError due to comparing a non-semver str against semver:

```
ERROR! Unexpected Exception, this is probably a bug: invalid semantic version '1.0'
the full traceback was:

Traceback (most recent call last):
  File "/home/liveuser/ansible/lib/ansible/cli/__init__.py", line 600, in cli_executor
    exit_code = cli.run()
  File "/home/liveuser/ansible/bin/ansible-galaxy", line 573, in run
    return context.CLIARGS['func']()
  File "/home/liveuser/ansible/bin/ansible-galaxy", line 90, in method_wrapper
    return wrapped_method(*args, **kwargs)
  File "/home/liveuser/ansible/bin/ansible-galaxy", line 1208, in execute_install
    self._execute_install_collection(
  File "/home/liveuser/ansible/bin/ansible-galaxy", line 1235, in _execute_install_collection
    install_collections(
  File "/home/liveuser/ansible/lib/ansible/galaxy/collection/__init__.py", line 513, in install_collections
    dependency_map = _resolve_depenency_map(
  File "/home/liveuser/ansible/lib/ansible/galaxy/collection/__init__.py", line 1329, in _resolve_depenency_map
    return collection_dep_resolver.resolve(
  File "/home/liveuser/.local/lib/python3.8/site-packages/resolvelib/resolvers.py", line 453, in resolve
    state = resolution.resolve(requirements, max_rounds=max_rounds)
  File "/home/liveuser/.local/lib/python3.8/site-packages/resolvelib/resolvers.py", line 347, in resolve
    failure_causes = self._attempt_to_pin_criterion(name, criterion)
  File "/home/liveuser/.local/lib/python3.8/site-packages/resolvelib/resolvers.py", line 216, in _attempt_to_pin_criterion
    satisfied = all(
  File "/home/liveuser/.local/lib/python3.8/site-packages/resolvelib/resolvers.py", line 217, in <genexpr>
    self._p.is_satisfied_by(r, candidate)
  File "/home/liveuser/ansible/lib/ansible/galaxy/dependency_resolution/providers.py", line 296, in is_satisfied_by
    return meets_requirements(
  File "/home/liveuser/ansible/lib/ansible/galaxy/dependency_resolution/versioning.py", line 62, in meets_requirements
    SemanticVersion(version),
  File "/home/liveuser/ansible/lib/ansible/utils/version.py", line 145, in __init__
    self.parse(vstring)
  File "/home/liveuser/ansible/lib/ansible/utils/version.py", line 194, in parse
    raise ValueError("invalid semantic version '%s'" % vstring)
ValueError: invalid semantic version '1.0'
```

I'll add a test for the second case.

Reply:
Added the second test.